pkg/ocrtypecodec/v1/exec.go

```diff
@@ -102,12 +108,19 @@ func (e *ExecCodecProto) DecodeOutcome(data []byte) (exectypes.Outcome, error) {
 		return exectypes.Outcome{}, fmt.Errorf("proto unmarshal ExecOutcome: %w", err)
 	}
 
+	var reports []cciptypes.ExecutePluginReport
+	if pbOutc.ExecutePluginReport != nil {
+		// Migrate legacy field by wrapping it in a slice and assigning it to the new field.
+		// TODO: Remove temporary migration code after a few releases.
+		reports = e.tr.execPluginReportsFromProto([]*ocrtypecodecpb.ExecutePluginReport{pbOutc.ExecutePluginReport})
+	} else {
+		reports = e.tr.execPluginReportsFromProto(pbOutc.ExecutePluginReports)
+	}
```

Comment:
The decode function wraps the legacy field in a slice so that old deployments produce outcomes that look like those expected by the new version.